### ISSUE TITLE
Reorder multithreaded job log before emailing it

### ISSUE THREAD

**Issue by wilprig** [rundeck/rundeck#1022]
Hello,

I have a job that executes a script on several nodes: the job is multhreaded and takes about 1h30  on each node. It sends the output log using the "attach output log" to some people that do not have access to rundeck (and they have no reason to).

My problem is that as the job is multithreaded, the log of each node is all mixed up (which is normal), which make it unreadable for the people receiving the mail (it can be done by grepping on each process id, but this is tedious...). So the solution i have for now is to not use multithreading which i don't like because it increases a lot the job duration just in the matter oh having a readable output

As it is possible to see the logs on each node via the web interface, i was thinking that rundeck could do the reordering of the full output log itself before sending it by mail (this could be an option of the job).

Thanks!

**Comment by gaddman** [comment]
Yeah, this would be great. The interleaved output of the log is useless if you want to run a job once, and then run again and do a diff comparison. I guess by the "Report" view you mean `http://localhost:4440/project/<projectname>/execution/show/<executionID>#nodes` ?
But good idea on disabling multithreading, I hadn't considered that as a workaround.

**Comment by fdevans** [comment]
Thanks for the report. As part of a cleanup of our issue tracker, we're closing issues older than 12 months, in line with Rundeck's support policy of one year from a release's date (see the [release calendar](https://docs.rundeck.com/docs/history/release-calendar.html) for currently supported versions). This isn't a judgment that the report is invalid — many of these are likely still real — but we can't verify against a version this old.

If you're still seeing this on a currently supported release (6.1.x or 5.20.x+), please open a new issue using our [bug report template](https://github.com/rundeck/rundeck/issues/new?template=bug_report.yml) with your version and reproduction steps. Please don't reopen this issue — a fresh issue keeps things tracked against a supported version and gets it in front of us properly. We're actively working on improving how issues get submitted and triaged going forward.

Thanks for your patience,
The Rundeck Team
